pandas_profiling/model/base.py

```diff
@@ -0,0 +1,224 @@
+    """Is the series boolean type?
+
+    Args:
+        series: Series
+        series_description: Series description
+
+    Returns:
+        True is the series is boolean type in the broad sense (e.g. including yes/no, NaNs allowed).
+    """
+    keys = series_description["value_counts_without_nan"].keys()
+    if pd.api.types.is_bool_dtype(keys):
+        return True
+    elif series_description[
+        "distinct_count_without_nan"
+    ] <= 2 and pd.api.types.is_numeric_dtype(series):
+        return True
+    elif series_description["distinct_count_without_nan"] <= 4:
+        unique_values = set([str(value).lower() for value in keys.values])
+        accepted_combinations = [["y", "n"], ["yes", "no"], ["true", "false"]]
+
+        if len(unique_values) == 2 and any(
+            [unique_values == set(bools) for bools in accepted_combinations]
+        ):
+            return True
+
+    return False
+
+
+def is_numeric(series: pd.Series, series_description: dict) -> bool:
+    """Is the series numeric type?
+
+    Args:
+        series: Series
+        series_description: Series description
+
+    Returns:
+        True is the series is numeric type (NaNs allowed).
+    """
+    return pd.api.types.is_numeric_dtype(series) and series_description[
+        "distinct_count_without_nan"
+    ] >= config["low_categorical_threshold"].get(int)
+
+
+def is_url(series: pd.Series, series_description: dict) -> bool:
+    """Is the series url type?
+
+    Args:
+        series: Series
+        series_description: Series description
+
+    Returns:
+        True is the series is url type (NaNs allowed).
+    """
+    if series_description["distinct_count_without_nan"] > 0:
+        try:
+            result = series[~series.isnull()].astype(str).apply(urlparse)
+            return result.apply(lambda x: all([x.scheme, x.netloc, x.path])).all()
+        except ValueError:
+            return False
+    else:
+        return False
+
+
+# def is_path(x):
+#     try:
+#         Path(x).is_file()
+#         return True
+#     except OSError:
+#         return False
+
+
+def _date_parser(date_string):
+    pattern = re.compile(r"[.\-:]")
+    pieces = re.split(pattern, date_string)
+
+    if len(pieces) < 3:
+        raise ValueError("Must have at least year, month and date passed")
+
+    return parse(date_string)
+
+
+def is_date(series) -> bool:
+    """Is the variable of type datetime? Throws a warning if the series looks like a datetime, but is not typed as
+    datetime64.
+
+    Args:
+        series: Series
+
+    Returns:
+        True if the variable is of type datetime.
+    """
+    is_date_value = pd.api.types.is_datetime64_dtype(series)
+    try:
+        series.apply(_date_parser)
+        warnings.warn(
+            'Column "{}" appears to be containing only date/datetime values. You might consider '
+            "changing the type to datetime (pd.to_datetime())".format(series.name)
+        )
+    except ValueError:
```

Comment:
cleaner way will be with ignore() context manager https://docs.python.org/3/library/contextlib.html

Reply:
Elegant, will change this.